lib/erb/formatter.rb

```diff
@@ -39,7 +39,7 @@ class Error < StandardError; end
   ATTR = Regexp.union(SINGLE_QUOTE_ATTR, DOUBLE_QUOTE_ATTR, UNQUOTED_ATTR, UNQUOTED_VALUE)
   MULTILINE_ATTR_NAMES = %w[class data-action]
 
-  ERB_TAG = %r{(<%(?:==|=|-|))\s*(.*?)\s*(-?%>)}m
+  ERB_TAG = %r{(<%(?:=|-|#)*)(?:(?!\n)\s)*(.*?)\s*(-?%>)}m
```

Comment:
fyi: `(?:(?!\n)\s)` matches all whitespace except for newlines. Done this in order to preserve comment indentation on the first line